src/include/stir/recon_buildblock/LogcoshPrior.h

```diff
@@ -224,18 +225,19 @@ class LogcoshPrior:  public
       { return tanh(x)/x; }
     }
 
-    //! The Hessian of log(cosh()) is sech^2(x) = (1/cosh(x))^2
+    //! The second partial derivatives of the LogCosh Prior
     /*!
-     This function returns the hessian of the logcosh function
-     * @param d the difference between the ith and jth voxel.
-     * @param scalar is the logcosh scalar value controlling the priors transition between the quadratic and linear behaviour
-     * @return the second derivative of the log-cosh function
+     Diagonal refers to the second derivative w.r.t. x_j only (i.e. diagonal of the Hessian)
+     Off-diagonal refers to the second derivative w.r.t. x_j and x_k (i.e. off-diagonal of the Hessian)
+     For LogCosh, the off diagonal is the negative of the diagonal.
+     * @param x_j is the target voxel.
+     * @param x_k is the voxel in the neighbourhood.
+     * @return the second order partial derivatives of the LogCosh Prior
```

Comment:
I don't understand the definition of these. I certainly can't get it from the documentation.

prior is a symmetric function of p(x_j,x_k). (many of them are a function of (x_j-x_k), but not all). Is "diagonal" the second derivative w.r.t. to the first arg, and "off_diagonal" the derivative wr.t. each arg? That would make sense to have. I'd then call them differently though. somethng like `derivative_20() `and `derivative_11`().

Reply:
Yes. diagonal refers to the partial derivatve `d^2/d x_j^2` and off-diagonal is `d^2/(d x_j d x_k)`. I can rename the functions `derivative_20` and `derivative_1` and detail why these names are used. We could also create a `derivative_1` at a later date in gradient computations.